### PR TITLE
Kill iperf3 processes when complete

### PR DESCRIPTION
**Pull request by swiftgist**
Signed-off-by: Eric Jackson <ejackson@suse.com>
bsc: 1122941
-----------------

**Checklist:**
- [x] Added unittests and or functional tests
- [ ] Adapted documentation
- [x] Referenced issues or internal bugtracker
- [ ] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

File: srv/salt/ceph/tests/iperf3/init.sls [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+starting iperf3:
+  module.run:
+    - name: multi.iperf_server_cmd
+
+check iperf3 up:
+  cmd.run:
+    - name: "pgrep iperf3"
```
Comment (jschmid1) [inline srv/salt/ceph/tests/iperf3/init.sls]: does this fail even without a 'failhard: True'?

**Comment by swiftgist** [comment]
done.

**Comment by jschmid1** [comment]
That'd be also a candidate for the 'functests.init'. Mind adding it @swiftgist

**Comment by jschmid1** [comment]
retest this please

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by kshtsk** [comment]
@jschmid1 this is artifacts issue I've fixed it

**Comment by kshtsk** [comment]
@susebot run teuthology

**Comment by kshtsk** [comment]
@susebot run teuthology

**Comment by kshtsk** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
A valid failure in the tests.

```
2019-02-19T14:18:57.849 INFO:teuthology.orchestra.run.target192168000082:Running: "sudo bash -c 'DEV_ENV=true timeout 60m salt-run --no-color state.orch ceph.functests.1node 2>/dev/null'"
2019-02-19T14:19:07.504 INFO:teuthology.orchestra.run.target192168000082.stdout:target192168000082.teuthology_master:
2019-02-19T14:19:07.505 INFO:teuthology.orchestra.run.target192168000082.stdout:    Data failed to compile:
2019-02-19T14:19:07.505 INFO:teuthology.orchestra.run.target192168000082.stdout:----------
2019-02-19T14:19:07.505 INFO:teuthology.orchestra.run.target192168000082.stdout:    Specified SLS ceph.functests.1node.multi in saltenv base is not available on the salt master or through a configured fileserver
```

**Comment by swiftgist** [comment]
That's not enough information to determine what data failed to compile.  And I cannot reproduce the problem in my environment.

**Comment by jschmid1** [comment]
What information would be required then?

**Comment by kshtsk** [comment]
I'm not a DeepSea developer but I can see in the logs following:
```
2019-02-19T14:21:24.032 INFO:teuthology.orchestra.run.target192168000060:Running: "sudo bash -c 'DEV_ENV=true timeout 60m deepsea --log-file=/var/log/salt/deepsea.log --log-level=debug salt-run state.orch ceph.functests.1node --simple-output'"
2019-02-19T14:21:24.507 INFO:teuthology.orchestra.run.target192168000060.stdout:Starting stage: ceph.functests.1node
2019-02-19T14:21:29.583 INFO:teuthology.orchestra.run.target192168000060.stdout:Parsing stage ceph.functests.1node steps... fail
2019-02-19T14:21:29.583 INFO:teuthology.orchestra.run.target192168000060.stdout:
2019-02-19T14:21:29.584 INFO:teuthology.orchestra.run.target192168000060.stdout:An error occurred while rendering the stage file:
2019-02-19T14:21:29.605 INFO:teuthology.orchestra.run.target192168000060.stderr:Traceback (most recent call last):
2019-02-19T14:21:29.605 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/bin/deepsea", line 9, in <module>
2019-02-19T14:21:29.605 INFO:teuthology.orchestra.run.target192168000060.stderr:    load_entry_point('deepsea==0.8.9+6.gebc30f5c', 'console_scripts', 'deepsea')()
2019-02-19T14:21:29.607 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/deepsea.py", line 371, in main
2019-02-19T14:21:29.607 INFO:teuthology.orchestra.run.target192168000060.stderr:    cli(prog_name='deepsea')
2019-02-19T14:21:29.607 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 722, in __call__
2019-02-19T14:21:29.607 INFO:teuthology.orchestra.run.target192168000060.stderr:    return self.main(*args, **kwargs)
2019-02-19T14:21:29.608 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 697, in main
2019-02-19T14:21:29.608 INFO:teuthology.orchestra.run.target192168000060.stderr:    rv = self.invoke(ctx)
2019-02-19T14:21:29.608 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
2019-02-19T14:21:29.609 INFO:teuthology.orchestra.run.target192168000060.stderr:    return _process_result(sub_ctx.command.invoke(sub_ctx))
2019-02-19T14:21:29.609 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 1066, in invoke
2019-02-19T14:21:29.611 INFO:teuthology.orchestra.run.target192168000060.stderr:    return _process_result(sub_ctx.command.invoke(sub_ctx))
2019-02-19T14:21:29.611 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 895, in invoke
2019-02-19T14:21:29.612 INFO:teuthology.orchestra.run.target192168000060.stderr:    return ctx.invoke(self.callback, **ctx.params)
2019-02-19T14:21:29.612 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/click/core.py", line 535, in invoke
2019-02-19T14:21:29.612 INFO:teuthology.orchestra.run.target192168000060.stderr:    return callback(*args, **kwargs)
2019-02-19T14:21:29.612 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/common.py", line 66, in func_wraper
2019-02-19T14:21:29.612 INFO:teuthology.orchestra.run.target192168000060.stderr:    return func(*args, **kwargs)
2019-02-19T14:21:29.613 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/deepsea.py", line 356, in state_orch
2019-02-19T14:21:29.613 INFO:teuthology.orchestra.run.target192168000060.stderr:    ret = run_stage(stage_name, hide_state_steps, hide_dynamic_steps, simple_output)
2019-02-19T14:21:29.613 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/stage_executor.py", line 79, in run_stage
2019-02-19T14:21:29.613 INFO:teuthology.orchestra.run.target192168000060.stderr:    mon.parse_stage(stage_name)
2019-02-19T14:21:29.613 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/monitor.py", line 538, in parse_stage
2019-02-19T14:21:29.614 INFO:teuthology.orchestra.run.target192168000060.stderr:    self._fire_event('stage_parsing_finished', None, None, ex)
2019-02-19T14:21:29.614 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/monitor.py", line 601, in _fire_event
2019-02-19T14:21:29.615 INFO:teuthology.orchestra.run.target192168000060.stderr:    getattr(listener, event)(*args)
2019-02-19T14:21:29.615 INFO:teuthology.orchestra.run.target192168000060.stderr:  File "/usr/lib/python2.7/site-packages/deepsea/monitors/terminal_outputter.py", line 50, in stage_parsing_finished
2019-02-19T14:21:29.615 INFO:teuthology.orchestra.run.target192168000060.stderr:    PP.println("    {}".format(exception.stage_file))
2019-02-19T14:21:29.615 INFO:teuthology.orchestra.run.target192168000060.stderr:AttributeError: 'StageRenderingException' object has no attribute 'stage_file'
```
(for details see http://storage-ci.suse.de:8080/job/pr-deepsea/23/testReport/teuthology/suse_tier1/functional__0_salt_yaml_1_deepsea_yaml_2_services_and_test_yaml_ceph_cm_salt_yaml_cluster__4disks_yaml_roles_yaml__deepsea_cli_on_yaml_distros_sle_12_3_yaml_/attachments/teuthology-8884.log)

**Comment by jschmid1** [comment]
That's the stacktrace of the deepsea-cli as a result of the actual orchestration command failing.

@rjfd may want to look into this

**Comment by jschmid1** [comment]
> What information would be required then?

@swiftgist bump

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
@susebot run teuthology

**Comment by jschmid1** [comment]
Same failure again @swiftgist. Seems reproducable, please have a look at the logs.

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by swiftgist** [comment]
@susebot run teuthology

**Comment by swiftgist** [comment]
@susebot run teuthology

**Pull request merged**